engine/common/provider/engine_test.go

```diff
@@ -165,6 +165,76 @@ func TestOnEntityRequestPartial(t *testing.T) {
 	con.AssertExpectations(t)
 }
 
+func TestOnEntityRequestDuplicates(t *testing.T) {
+
+	entities := make(map[flow.Identifier]flow.Entity)
+
+	identities := unittest.IdentityListFixture(8)
+	selector := filter.HasNodeID(identities.NodeIDs()...)
+	originID := identities[0].NodeID
+
+	coll1 := unittest.CollectionFixture(1)
+	coll2 := unittest.CollectionFixture(2)
+	coll3 := unittest.CollectionFixture(3)
+
+	entities[coll1.ID()] = coll1
+	entities[coll2.ID()] = coll2
+	entities[coll3.ID()] = coll3
+
+	retrieve := func(entityID flow.Identifier) (flow.Entity, error) {
+		entity, ok := entities[entityID]
+		if !ok {
+			return nil, storage.ErrNotFound
+		}
+		return entity, nil
+	}
+
+	final := &protocol.Snapshot{}
```

Comment:
```suggestion
	final := protocol.NewSnapshot(t)
```
Let's always use the default provided `testify` mock constructor that checks for expectations automatically.

Reply:
https://github.com/onflow/flow-go/pull/3270/commits/226016109bb39372228bb31964a928ecafac5f1e